Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp

```diff
@@ -6,8 +6,155 @@
  *
  */
 
-#include <AzCore/Memory/AllocatorManager.h>
 #include <AzCore/Memory/Memory.h>
+#include <AzCore/Memory/AllocatorManager.h>
+
+#define RECORDING_ENABLED 0
+
+#if RECORDING_ENABLED
+
+#include <AzCore/std/containers/map.h>
+#include <AzCore/IO/SystemFile.h>
+#include <AzCore/std/parallel/mutex.h>
+#include <AzCore/std/parallel/scoped_lock.h>
+
+namespace
+{
+    class DebugAllocator
+    {
+    public:
+        typedef void* pointer_type;
```

Comment:
Nit: prefering `using` syntax (e.g. `using pointer_type = void*;`) for these declarations (more modern, reads left-to-right, etc.)

Reply:
sure, will change, this was some code I took from a very old implementation I had around